platform/android/java/app/build.gradle

```diff
@@ -58,6 +58,9 @@ dependencies {
     if (pluginsBinaries != null && pluginsBinaries.size() > 0) {
         implementation files(pluginsBinaries)
     }
+
+    // .NET dependencies
+    implementation files('../../../../modules/mono/thirdparty/libSystem.Security.Cryptography.Native.Android.jar')
```

Comment:
I don't like including the .jar like this. This .jar comes from the .NET runtime pack, but we don't have it at the time of building the export template and adding a .jar when exporting the project doesn't seem to be possible without forcing the user to enable _Use Gradle Build_.

Reply:
In addition to the comment about using `flavor` to guard the dependency, the jar file also needs to be within the `app` directory otherwise this will break editor gradle builds.

Reply:
For gradle builds we probably want to get the jar file from the built project. But for now I can just move the jar to the `app` directory, should I put it in any specific subdirectory?

Reply:
it would typically go into the `libs` directory, but the contents of that directory get deleted during a clean build, so let's create a new `monoLibs` directory and include the jar there.

> For gradle builds we probably want to get the jar file from the built project

How would that work? Does the user project include the jar file?

Reply:
> How would that work? Does the user project include the jar file?

When the C# project is built, during the restore step it will retrieve all the dependencies packages. For android builds, this means it will retrieve the Mono runtime package (e.g.: [Microsoft.NETCore.App.Runtime.Mono.android-arm](https://www.nuget.org/packages/Microsoft.NETCore.App.Runtime.Mono.android-arm) for `android-arm32`). This package contains the jar file for the runtime your project targets and we can retrieve it from your NuGet packages.

Reply:
So the build process include fetching remote dependencies which include the jar file?

Is it then possible to configure where the dependency is stored, so we can have it add / overwrite the jar file included in the `monoLibs` directory?

Reply:
We can copy the jar file as part of the build process.

Reply:
If that's the case, then we don't need to include the jar file into the `monoLibs`.

There will be two routes, both gated by the `mono` flavor:
- If generating the build template from within the Godot code base, then we can check if the jar file exists and only then add a dependency to it. You can see the same approach [here](https://github.com/godotengine/godot/blob/29b3d9e9e538f0aa8effc8ad8bf19a2915292a89/platform/android/java/app/build.gradle#L36-L46) where we check which version of the Godot library is available before adding the dependency.

- If generating as a gradle build from within the Godot Editor, we can use logic similar to the one described above to check for the presence of the jar file in the local `libs` directory. The jar file would have been copied there by the conclusion of the c# project build process as you described.

Reply:
I'm not sure I understand how flavors work, how do I _enable_ the `mono` flavor when building the templates?

Reply:
We use the `assemble` gradle task for building and it's structured like this: `assemble[flavors...][buildtypes...]`.
So by default if you only specify the `assemble` task, it builds all the flavors and build types; if you specify the build type (e.g: `assembleDebug`), then it builds all the flavors; and so on.. So if you don't change anything, the `mono` flavor will be included as part of the build, and if you only want to build the `mono` flavor, you can add it as a prefix to the `assemble` task.

After you configure the flavor, you can use the [Gradle tool window](https://www.jetbrains.com/help/idea/jetgradle-tool-window.html#gradle_tool_window) (or the [following command](https://stackoverflow.com/a/40188539)) to see the tasks that are being added.

> And I guess I have to add a standard flavor for the non-mono version?

Yes you would need to add a `standard` flavor as an alternative to the `mono` flavor.

>  Do I have to duplicate every copy{Target}BinaryToBin task for the mono flavor?

As currently setup, you would need to duplicate the copy tasks. Let me take a look and see if I can clean that setup.

Reply:
> Let me take a look and see if I can clean that setup.

@raulsntos I've cleaned the gradle build setup in https://github.com/godotengine/godot/pull/91271. Feel free to comment on the PR if you have questions on the update.

Reply:
Looks great, is something like https://github.com/raulsntos/godot/commit/ad0f8fec199173d0441f3177bd81bd87d37ea101 what you had in mind? Or should the `generateBuildTasks` function take an extra `edition` parameter to only include a single edition build task? If so, should I create a new `generateGodotTemplatesMono` task for this or can the `generateGodotTemplates` task use parameters?

Reply:
Let's go with the second approach and have `generateBuildTasks` take an extra `edition` parameter and add a `generateGodotMonoTemplates` task that invokes it.

That matches our current [build setup](https://github.com/godotengine/godot-build-scripts/blob/main/build-android/build.sh), where we have separate sections for standard and mono builds. And it allows users to only build the edition they need.

cc @akien-mga as this'll affect the android build scripts.

Reply:
Made a PR in the `godot-build-scripts` repo:

- https://github.com/godotengine/godot-build-scripts/pull/94